nixos/modules/programs/zsh/zsh.nix

```diff
@@ -67,9 +67,7 @@ in
       };
 
       promptInit = mkOption {
-        default = ''
-          autoload -U promptinit && promptinit && prompt walters
```

Comment:
right, I think adding this will have the expected behaviour while keeping walters the default for the promptInit option.

```nix
{
  config = {
    # or maybe? (cfg.oh-my-zsh.theme != "")
    programs.zsh.promptInit = mkIf cfg.oh-my-zsh.enable (mkDefault "");
  };
}
```